src/modules/authorizer/role/AUT_TokenGated_Roles_v1.sol

```diff
@@ -92,39 +92,62 @@ contract AUT_TokenGated_Roles_v1 is IAUT_TokenGated_Roles_v1, AUT_Roles_v1 {
     //--------------------------------------------------------------------------
     // Overloaded and overriden functions
 
-    function hasRole(bytes32 roleId, address account)
+    function hasRole(bytes32 roleId, address who)
         public
         view
         virtual
         override(AccessControlUpgradeable, IAccessControl)
         returns (bool)
     {
         if (isTokenGated[roleId]) {
-            return _hasTokenRole(roleId, account);
+            return _hasTokenRole(roleId, who);
         } else {
-            return super.hasRole(roleId, account);
+            return super.hasRole(roleId, who);
         }
     }
 
     /// @notice Grants a role to an address
     /// @param role The role to grant
-    /// @param account The address to grant the role to
+    /// @param who The address to grant the role to
     /// @return bool Returns if the role has been granted succesful
     /// @dev Overrides {_grantRole} from AccesControl to enforce interface implementation when role is token-gated
-    function _grantRole(bytes32 role, address account)
+    function _grantRole(bytes32 role, address who)
         internal
         virtual
         override
         returns (bool)
     {
         if (isTokenGated[role]) {
-            try TokenInterface(account).balanceOf(address(this)) {}
+            // Make sure that a threshold has been set before granting the role
+            if (getThresholdValue(role, who) == 0) {
+                revert Module__AUT_TokenGated_Roles__TokenRoleMustHaveThreshold(
+                    role, who
+                );
+            }
+            try TokenInterface(who).balanceOf(address(this)) {}
             catch {
-                revert Module__AUT_TokenGated_Roles__InvalidToken(account);
+                revert Module__AUT_TokenGated_Roles__InvalidToken(who);
             }
         }
 
-        return super._grantRole(role, account);
+        return super._grantRole(role, who);
+    }
+
+    /// @notice Overrides {_revokeRole} to prevent having an empty OWNER role
```

Comment:
Is this comment valid? Doesn't this just prevent the leftover thresholds after removing the role? Or am I missing something?

Reply:
You are right. The parent function does, but I updated the comment